src/complex/DataStructure/AbstractDataStore.hpp

```diff
@@ -506,12 +509,145 @@ class AbstractDataStore : public IDataStore
    * @param i
    * @param value
    */
-  void fillTuple(usize i, T value)
+  void fillTuple(index_type i, T value)
   {
     usize numComponents = getNumberOfComponents();
     std::fill_n(begin() + (i * numComponents), numComponents, value);
   }
 
+  /**
+   * @brief Sets all component values for a tuple using a pointer array of values.
+   * The provided pointer is expected to contain at least the same number of values
+   * as the number of components.
+   *
+   * If the tuple index is out of bounds or the provided pointer is null, this method does nothing.
+   * @param tupleIndex
+   * @param values
+   */
+  void setTuple(index_type tupleIndex, value_type* values)
+  {
+    if(values == nullptr)
+    {
+      return;
+    }
+
+    if(tupleIndex > getNumberOfTuples())
+    {
+      return;
+    }
+
+    index_type numComponents = getNumberOfComponents();
+    index_type offset = tupleIndex * numComponents;
+    for(index_type i = 0; i < numComponents; i++)
+    {
+      setValue(offset + i, values[i]);
+    }
+  }
+
+  /**
+   * @brief Sets all component values for a tuple using a vector of values.
+   *
+   * If the tuple index is out of bounds or the provided vector does not match
+   * the number of components, this method does nothing.
+   * @param tupleIndex
+   * @param values
+   */
+  void setTuple(index_type tupleIndex, const std::vector<value_type>& values)
+  {
+    if(values.size() != getNumberOfComponents())
+    {
+      return;
+    }
+
+    if(tupleIndex > getNumberOfTuples())
+    {
+      return;
+    }
+
+    index_type numComponents = getNumberOfComponents();
+    index_type offset = tupleIndex * numComponents;
+    for(index_type i = 0; i < numComponents; i++)
+    {
+      setValue(offset + i, values[i]);
+    }
+  }
+
+  /**
+   * @brief Sets all component values for a tuple using a span of values.
+   *
+   * If the tuple index is out of bounds or the provided span does not match
+   * the number of components, this method does nothing.
+   * @param tupleIndex
+   * @param values
+   */
+  void setTuple(index_type tupleIndex, const nonstd::span<value_type>& values)
```

Comment:
```suggestion
  void setTuple(index_type tupleIndex, nonstd::span<const value_type> values)
```
We really only need span since a span can be constructed from a pointer, vector, etc.